File: typescript-connector/src/models/ProductOptionValue.ts
// Class from DFC Business Ontology: #ProductOptionValue
import { SemanticObject } from "../core/SemanticObject.js";
import { WhatSubject, type WhatSubjectParams } from "./WhatSubject.js";

export interface ProductOptionValueParams extends WhatSubjectParams {}
export class ProductOptionValue extends WhatSubject {
  static get SEMANTIC_TYPE(): string {
    return "dfc-b:ProductOptionValue";
  }



  constructor(
    semanticId: string,
    params?: ProductOptionValueParams,
  ) {
    super(semanticId, params);
    this.semanticType = ProductOptionValue.SEMANTIC_TYPE;
  }
  static {
    SemanticObject.typeRegistry.set(ProductOptionValue.SEMANTIC_TYPE, ProductOptionValue);
  }
}

File: ruby-gem/lib/models/product_option_value.rb
# frozen_string_literal: true

# Class from DFC Business Ontology: #ProductOptionValue
require_relative '../core/semantic_object'
require_relative 'what_subject'


module DfcLinkmlConnector
  module Models
    class ProductOptionValue < WhatSubject
      SEMANTIC_TYPE = "dfc-b:ProductOptionValue".freeze

      # @param semanticId [String]
      # @param date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil
      def initialize(semanticId, date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil)
        super(semanticId, date: date, description: description, name: name, characteristicOf: characteristicOf, dimension: dimension)

        self.semanticType = "dfc-b:ProductOptionValue"

      end

      Core::SemanticObject.type_registry[SEMANTIC_TYPE] = self
    end
  end
end
